### PR TITLE
Implements architecture rspec tests

### PR DESCRIPTION
**Pull request by ChrisBr**

### CODE REVIEW

File: src/api/spec/controllers/webui/architectures_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
+require 'rails_helper'
+
+RSpec.describe Webui::ArchitecturesController do
+  let(:confirmed_user) { create(:confirmed_user) }
+  let(:admin_user) { create(:admin_user) }
+
+  describe 'GET #index' do
```
Comment (hennevogel) [inline src/api/spec/controllers/webui/architectures_controller_spec.rb]: I don't like this. We should test that this controller has the require_admin before_filter and that Webui::WebuiController#require_admin works

**Comment by ChrisBr** [comment]
@hennevogel please review again. Test now if the controller uses the require_admin before filter!

**Comment by hennevogel** [comment]
And who tests the before filter? And why do you still test the admin case?

**Comment by ChrisBr** [comment]
@hennevogel before_filter is tested in #1532

**Comment by ChrisBr** [comment]
@hennevogel I don't test the admin case, I test the functionality of the controller (which is only possible when I'm an admin)... 
We can discuss if renaming of the test case makes sense ...

**Comment by hennevogel** [comment]
You test that  get index returns 200. Whats the value of that?

**Comment by hennevogel** [comment]
Cool!

**Pull request merged**